File: src/main/java/io/github/vpavic/traintracker/config/SessionConfiguration.java
package io.github.vpavic.traintracker.config;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.session.jdbc.PostgreSqlJdbcIndexedSessionRepositoryCustomizer;

@Configuration(proxyBeanMethods = false)
class SessionConfiguration {

    @Bean
    PostgreSqlJdbcIndexedSessionRepositoryCustomizer sessionRepositoryCustomizer() {
        return new PostgreSqlJdbcIndexedSessionRepositoryCustomizer();
    }

}

File: src/main/kotlin/io/github/vpavic/traintracker/config/SessionConfiguration.kt
package io.github.vpavic.traintracker.config

import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.session.jdbc.PostgreSqlJdbcIndexedSessionRepositoryCustomizer

@Configuration(proxyBeanMethods = true)
class SessionConfiguration {

    @Bean
    fun sessionRepositoryCustomizer(): PostgreSqlJdbcIndexedSessionRepositoryCustomizer {
        return PostgreSqlJdbcIndexedSessionRepositoryCustomizer()
    }
}
